test/runtests.jl

```diff
@@ -44,6 +44,7 @@ test_pattern_get_surface()
 pth = joinpath(dirname(dirname(@__FILE__)), "samples")
 fls = filter(str->endswith(str,".jl"), readdir(pth))
 for fl in fls
+    println(fl)
```

Comment:
Rather than an IO buffer, try it first with a different kind of stream instead:
```jl
img = open(filename) do io
    read_from_png(io)
end
```
It's quite possible that the Cairo library won't support julia's IOBuffer type, which is not a stream in the Unix sense (C libraries are oh-so-picky). Note:
```jl
julia> supertype(typeof(STDIN))
Base.LibuvStream

julia> supertype(supertype(typeof(STDIN)))
IO

julia> supertype(IOBuffer)
IO
```